src/Agent.Worker/Worker.cs

```diff
@@ -185,11 +185,11 @@ private void InitializeSecretMasker(Pipelines.AgentJobRequestMessage message)
             // Add masks for service endpoints
             foreach (ServiceEndpoint endpoint in message.Resources.Endpoints ?? new List<ServiceEndpoint>())
             {
-                foreach (string value in endpoint.Authorization?.Parameters?.Values ?? new string[0])
+                foreach (var keyValuePair in endpoint.Authorization?.Parameters)
```

Comment:
This _foreach_ statement will throw NullRef exception if Parameters will be null. You need either to do something like it was previously when the null value was replaced by the empty array or you can check Parameters for null before entering foreach.

Reply:
Fixed thanks!